coast/CLIMATOLOGY.py

```diff
@@ -52,15 +57,18 @@ def make_climatology(
         """
 
         frequency_str = time_var_name + "." + output_frequency
-        print("Calculating climatological mean")
+        info("Calculating climatological mean")
 
         if missing_values:
             ds_mean = xr.Dataset()
-            for varname, da in ds.data_vars.items():
-                mask = xr.where(uf.isnan(da), 0, 1)
-                data = da.groupby(frequency_str).sum(dim=time_dim_name)
-                N = mask.groupby(frequency_str).sum(dim=time_dim_name)
-                ds_mean[varname] = data / N
+            for var_name, da in ds.data_vars.items():
+                try:
+                    mask = xr.where(uf.isnan(da), 0, 1)
+                    data = da.groupby(frequency_str).sum(dim=time_dim_name)
+                    total_points = mask.groupby(frequency_str).sum(dim=time_dim_name)
+                    ds_mean[var_name] = data / total_points
+                except Exception as e:
+                    error(f"Problem with {var_name}: {e}")
```

Comment:
I know it's a faff, but could we look into catching specific errors here?

Additionally, it would be preferable to log the full error text. You can use `traceback.format_exc()` to get it as a string.

Reply:
Have updated this now. 

I ended up simplify that whole method which was a method that already existed. It now uses xarrays built in mean() method, with the skipna flag. 
The whole missing_values section has now been removed.